text/0000-semicolon-items.md

```diff
@@ -0,0 +1,448 @@
+# Semicolon Items
+
+- Feature Name: `semicolon_items`
+- Start Date: 2018-06-17
+- RFC PR:
+- Rust Issue:
+
+# Summary
+[summary]: #summary
+
+The semicolon (`;`) is now legal, but not recommended, where an item is
+expected, permitting a user to write `struct Foo {};`, among other things.
+The `item` fragment specifier in a `macro_rules` matcher will match `;`.
+To retain a recommended and uniform style, the tool `rustfmt` will remove any
+extraneous `;`. Furthermore, the compiler will fire a warn-by-default lint when
+extraneous `;` are encountered, whether they be inside or outside an `fn` body.
+
+# Motivation
+[motivation]: #motivation
+
+## Inform the user of multiple mistakes at once
+
+Consider the following program which according to current Rust compilers is
+illegal:
+
+```rust
+struct Foo {};
+struct Bar {};
+struct Baz {};
+
+fn main() {}
+```
+
+when type checking this with `cargo check`, you will first get the error:
+
+```rust
+1 | struct Foo {};
+  |              ^ help: consider removing this semicolon
+```
+
+which you quickly fix and `check` again, receiving:
```

Comment:
This is not a motivation for accepting semicolons, this is a motivation for doing better error recovery so that all semicolon-related errors are reported at once and there's no "`check` again, receive more errors" step.

Reply:
This section, in context is motivation, since the next paragraph talks about `cargo test`, which you will not get to do if your program does not compile.